src/Plugin/DocReader/DocReaderInterface.php

```diff
@@ -0,0 +1,7 @@
+<?php
+
+namespace Drupal\dropai\Plugin\DocReader;
+
+interface DocReaderInterface {
```

Comment:
Suggestion: Whenever possible please use type hinting for parameters and return types. Also add the related comments to document this so our IDE can do type checking.